seisbench/models/base.py

```diff
@@ -745,237 +1076,485 @@ async def _process_streams_to_arrays(
-                    pred_start = int(start * prediction_sample_factor)
-                    pred_merge[
-                        pred_start : pred_start + pred.shape[0], :, i % coverage
-                    ] = pred
-
-                with warnings.catch_warnings():
-                    warnings.filterwarnings(
-                        action="ignore", message="Mean of empty slice"
-                    )
-                    preds = np.nanmean(pred_merge, axis=-1)
+    def _reassemble_blocks_point(self, elem, buffer, argdict):
+        """
+        Reassembles point predictions into numpy arrays. Returns None except if a buffer was processed.
+        """
+        stride = argdict.get("stride", 1)
+
+        window, metadata = elem
+        t0, s, len_starts, trace_stats, bucket_id = metadata
+        key = f"{t0}_{trace_stats.network}.{trace_stats.station}.{trace_stats.station}.{trace_stats.channel[:-1]}"
+
+        output = None
+
+        buffer[key].append(elem)
+        if len(buffer[key]) == len_starts:
+            preds = [(s, window) for window, (_, s, _, _, _) in buffer[key]]
+            preds = sorted(
+                preds, key=lambda x: x[0]
+            )  # Sort by start (overwrite keys to make sure window is never used as key)
+            preds = [window for s, window in preds]
+            preds = np.stack(preds, axis=0)
+            if preds.ndim == 1:
+                preds = preds.reshape(-1, 1)
+            pred_time = t0 + self.pred_sample / argdict["sampling_rate"]
+            pred_rate = argdict["sampling_rate"] / stride
+
+            output = ((pred_rate, pred_time, preds), trace_stats)
+
+            del buffer[key]
+
+        return output
+
+    @log_lifecycle(logging.DEBUG)
+    def _process_cut_fragments_array(
+        self, queue_watchdog, queue_in, queue_out, argdict
+    ):
+        """
+        Wrapper with queue IO functionality around :py:func:`_cut_fragments_array`
+        """
+        while True:
+            elem = queue_in.get()
+            queue_in.task_done()
+            if elem is None:
+                break
+
+            for output_elem in self._cut_fragments_array(elem, argdict):
+                queue_out.put(output_elem)
+
+        queue_watchdog.put("cut")
+
+    async def _async_cut_fragments_array(self, queue_in, queue_out, argdict):
+        """
+        Wrapper with queue IO functionality around :py:func:`_cut_fragments_array`
+        """
+        while True:
+            elem = await queue_in.get()
+            if elem is None:
+                break
+
+            for output_elem in self._cut_fragments_array(elem, argdict):
+                await queue_out.put(output_elem)
+
+    def _cut_fragments_array(self, elem, argdict):
+        """
+        Cuts numpy arrays into fragments for array prediction models.
+        """
+        overlap = argdict.get("overlap", 0)
+
+        t0, block, trace_stats = elem
+
+        bucket_id = np.random.randint(1000000)
```

Comment:
We can boost this value all the way to 1e9 without affecting compute time, to lower probably of collisions. It is still probably not the optimum way of ensuring there are no collisions, but this approach provides the quickest way of assigning bucket-ids with v. low probability of the id matching.

Reply:
I'm fine with the value either way. I guess we are anyhow so close to uniformity over the workers (even if people go up to something like 100) that it won't really matter. Alternatively, one could generate the `bucket_id` by hashing some key, but there's not really an advantage in this and on the downside one needs to identify a good key to hash.

Reply:
Increased the value to 1e9 now.